src/pages/Post/Post.tsx

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable prettier/prettier */
+import './Post.css'
+
+import axios from 'axios'
+import React, {useEffect, useState } from 'react'
+
+interface IPost {
+    id: number,
+    links: {id: number, caption: string, url: string}[],
+    caption: string, 
+    short_text: string, 
+    details: string,
+    added_at: string,
+    show_after: string,
+    disable_after: string
+}
+
+const defaultPosts: IPost[] = []
+
+export const Posts: React.FC = () => {
+    const [posts, setPosts]: [IPost[], (posts: IPost[]) => void] = useState(defaultPosts)
+    const [loading, setLoading]: [boolean, (loading: boolean) => void] = React.useState<boolean>(true)
+    const [error, setError]: [string, (error: string) => void] = React.useState("")
+    
+    useEffect(() => {
+        axios.get<IPost[]>('/api/cms/post/visible/', {
+            headers: {
+                "Content-type": "application/json"
+            }
+        })
+        .then(response => {
+            setPosts(response.data)
+            setLoading(false)
+        })
+        .catch(ex => {
+            const error =
+            ex.response.status === 404
+                ? "Resource Not found"
+                : "An unexpected error has occurred"
+            setError(error)
+            setLoading(false)
+        })

```

Comment:
skus prosim prepisat do try-catch formatu, skusime ist konzistentne tym smerom. struktura bude:
```js
try {
  const response = await axios.get(...)
  setPosts(response.data)
} catch (ex) {
  ...
} finally {
  setLoading(false)
}
```
dokonca rozumne pouzit `finally` pre ukoncenie loadingu, ci uz skonci request uspesne, alebo nie

Reply:
jo a kedze je to v useEffecte, tak ti sam ESLint potom tusim ponukne moznost, ako to prepisat pomocou `fetchData` funkcie (nepaci sa mu velmi `await` len tak v effecte)

Reply:
prepisal som naspat do `axios.get` a `async/await` syntaxe, aj otypoval error